lib/src/model/nutriments.dart

```diff
@@ -43,21 +58,53 @@ class Nutriments extends JsonObject {
   ) =>
       '${nutrient.offTag}_${perSize.offTag}';
 
+  /// Returns the modifier key for a [nutrient]
+  String _getModifierTag(final Nutrient nutrient) =>
+      '${nutrient.offTag}_modifier';
+
   /// Returns the value in grams of that [nutrient] for that [perSize].
   ///
   /// It won't be grams for very specific nutrients, like [Nutrient.alcohol].
   double? getValue(final Nutrient nutrient, final PerSize perSize) =>
-      _map[_getTag(nutrient, perSize)];
+      _valueMap[_getTag(nutrient, perSize)];
+
+  /// Returns the modifier for a [nutrient].
+  NutrientModifier? getModifier(final Nutrient nutrient) =>
+      _modifierMap[nutrient.offTag];
 
   /// Sets the value in grams of that [nutrient] for that [perSize].
   ///
   /// It won't be grams for very specific nutrients, like [Nutrient.alcohol].
   Nutriments setValue(
     final Nutrient nutrient,
     final PerSize perSize,
-    final double? value,
-  ) {
-    _map[_getTag(nutrient, perSize)] = value;
+    final double? value, {
+    final NutrientModifier? modifier,
+  }) {
+    if (modifier == NutrientModifier.notProvided) {
+      if (value != null) {
+        throw Exception(
+          'When the modifier is set to "NutrientModifier.notProvided", the value must be null.',
+        );
+      }
+
+      /// Reset all values
+      for (final PerSize perSize in PerSize.values) {
+        _valueMap.remove(_getTag(nutrient, perSize));
+        _valueMap.remove(_getTag(nutrient, perSize));
+      }
+      _modifierMap[nutrient.offTag] = modifier;
+      return this;
+    }
+
+    if (modifier != null && value == null) {
+      throw Exception(
+        'When the modifier is set, the value must not be null.',
+      );
+    }
+
+    _valueMap[_getTag(nutrient, perSize)] = value;
+    _modifierMap[nutrient.offTag] = modifier;
```

Comment:
Could a modifier be different for different `PerSize`s?

Reply:
I'm not sure, but I can support it